sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -99,9 +102,18 @@ static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
 #if defined(MODULE_OD) && ENABLE_DEBUG
         od_hex_dump(hdr, nread, OD_WIDTH_DEFAULT);
 #endif
-
-        gnrc_pktbuf_remove_snip(pkt, eth_hdr);
-        LL_APPEND(pkt, netif_hdr);
+        /* if pkt has payload */
+        if (pkt != eth_hdr) {
+            /* replace IEEE 802.15.4 header with generic netif header */
```

Comment:
This PR needs rework anyway. So thanks, but this line will probably disappear ;-)